### PR TITLE
[Azure AMQP] Websocket Heartbeats & Retries

### PR DESCRIPTION
**Pull request by kashifkhan**
This PR addresses an issue that came over with a customer using Service Bus w/ websockets & autolock renewer

Using the receiver iterator and an auto lock renewer, if there was a pause (or busy function) of over one minute before completing the message, an error `Cannot write to closing transport` shows up in the logs. 

This error was being raised by aiohttp websocket library as the heartbeat was not received from the service within 10 seconds ( the default)

The Flow of the Issue

* As the work was happening (60 seconds or higher), keep alive would start in the background ( empty frame every 30 seconds )
* The websocket library has also started a heartbeat in the background waiting for some response within 10 seconds 
* As no heartbeat was received within 10 seconds, the library says "looks like the connection dropped, close the underlying transport"
* 20 seconds later keep alive would try to send an empty frame and would end up erroring, outputting the message above. 

The Fix
* Increase the heartbeat to be a little longer, at 40 seconds. This allows heartbeat to go out and make a call in the interim keeping things active

As I was checking side effects with regards to network drops I noticed that retries were not happening during a connection drop with websockets. I added in the exceptions that can be raised to address this as well. 

Some misc thoughts

* Do we still need heartbeat with keep alive?
* *  we do, heartbeat was added in to work around an issue in aiohttp websockets where it would react really slowly when a connection would drop & would end up hanging. This would pick up a drop much faster. The bugs reports are commented in the code
* the sync library reacts very quickly and doesn't have this problem.

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
hoping stress tests w/ network drops go smoothly 🤞

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/constants.py [hunk 1]
```diff
@@ -85,7 +85,7 @@
 AUTH_TYPE_SASL_PLAIN = "AUTH_SASL_PLAIN"
 AUTH_TYPE_CBS = "AUTH_CBS"
 
-DEFAULT_WEBSOCKET_HEARTBEAT_SECONDS = 10
+DEFAULT_WEBSOCKET_HEARTBEAT_SECONDS = 40
```
[the remark this answers is not in the document]
Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/constants.py:88]: 35 would work as well, I went a little higher to see if things get too messy with keep alive failing and heartbeat kicking in very closely. This gives a little room for keep alive to fail first and immediately try reconnecting. We can play around with different values

[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/constants.py:88]: I vote for 36.7.   There's too much focus on these round numbers.   We need more chaos.

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py [hunk 2]
```diff
@@ -569,7 +569,11 @@ async def _read(self, toread, buffer=None, **kwargs):  # pylint: disable=unused-
                     self._read_buffer = BytesIO(data[toread:])
                     toread = 0
             return view
-        except:
+        except TypeError as te:
+            # aiohttp websocket raises a TypeError when a websocket disconnects, as it ends up
+            # reading None over the wire and cant convert that to bytes
+            raise ConnectionError('Websocket disconnected: %r' % te)
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py:575]:
```suggestion
            raise ConnectionError('Websocket disconnected: %r' % te) from None
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/constants.py (hunk 1 above)
Comment (richardpark-msft) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/constants.py:88]:
```suggestion
DEFAULT_WEBSOCKET_HEARTBEAT_SECONDS = 36.7
```

**Comment by richardpark-msft** [comment]
@kashifkhan, I'm not sure this is the right approach to take. I'd close this PR and start again.

**Pull request closed without merging**